DFC.ServiceTaxonomy.ApiFunction.Tests/ExecuteHttpTriggerTests.cs

```diff
@@ -0,0 +1,251 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Castle.Components.DictionaryAdapter;
+using DFC.ServiceTaxonomy.ApiFunction.Function;
+using DFC.ServiceTaxonomy.ApiFunction.Helpers;
+using DFC.ServiceTaxonomy.ApiFunction.Models;
+using FakeItEasy;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Internal;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Neo4j.Driver.V1;
+using Newtonsoft.Json;
+using Xunit;
+using ILogger = Microsoft.Extensions.Logging.ILogger;
+
+namespace DFC.ServiceTaxonomy.ApiFunction.Tests
+{
+    public class ExecuteHttpTriggerTests
+    {
+        private readonly Execute _executeFunction;
+        private readonly ILogger _log;
+        private readonly HttpRequest _request;
+        private IOptions<ServiceTaxonomyApiSettings> _config;
+        private IHttpRequestHelper _httpRequestHelper;
+        private IJsonHelper _jsonHelper;
+        private INeo4JHelper _neo4JHelper;
+        private IFileHelper _fileHelper;
+        private Cypher _blankCypherModel = null;
+        private Cypher _cypherModel;
+
+        public ExecuteHttpTriggerTests()
+        {
+            _request = new DefaultHttpRequest(new DefaultHttpContext());
+            var options = new ServiceTaxonomyApiSettings
+            {
+                Function = "GetAllSkills",
+                Neo4jUrl = "bolt://localhost:11002",
+                Neo4jUser = "NeoUser",
+                Neo4jPassword = "NeoPass"
+            };
+            _config = Options.Create(options);
+            _log = A.Fake<ILogger>();
+            _httpRequestHelper = A.Fake<IHttpRequestHelper>();
+            _jsonHelper = A.Fake<IJsonHelper>();
+            _neo4JHelper = A.Fake<INeo4JHelper>();
+            _fileHelper = A.Fake<IFileHelper>();
+            _cypherModel = new Cypher {Query = "query", QueryParam = new List<QueryParam>{ new QueryParam { Name = "occupation" } } };
+
+            _executeFunction = new Execute(_config, _httpRequestHelper, _jsonHelper, _neo4JHelper, _fileHelper);
+
+        }
+
+        [Fact]
+        public async Task Execute_ReturnsBadRequestObjectResult_WhenFunctionAppSettingIsNullOrEmpty()
+        {
+            _config.Value.Function = null;
+
+            var result = await RunFunction();
+
+            var badRequestObjectResult = result as BadRequestObjectResult;
+
+            // Assert
+            Assert.IsAssignableFrom<IActionResult>(result);
+            Assert.Equal((int?)HttpStatusCode.BadRequest, badRequestObjectResult.StatusCode);
```

Comment:
It'd be nice to cover the case if `result` isn't convertible to `BadRequestObjectResult`, with an explicit `Assert`, rather than the test throwing a `NullReferenceException`.